collections_hierarchy/mixins.py

```diff
@@ -1,87 +1,98 @@
 class ComparableMixin(object):
     def __eq__(self, other):
-        pass
+        return self == other
 
     def __ne__(self, other):
         return not self == other
 
     def __lt__(self, other):
-        pass
+        return len(self) < len(other)
 
     def __gt__(self, other):
-        pass
+        return len(self) > len(other)
 
     def __le__(self, other):
-        pass
+        return len(self) <= len(other)
 
     def __ge__(self, other):
-        pass
+        return len(self) >= len(other)
 
 
 class SequenceMixin(object):
     def __iter__(self):
-        pass
+        return iter(self)
 
     def __next__(self):
-        pass
+        if isinstance(self.values, list):
+            if self.i > len(self.values) - 1:
+                raise StopIteration
+            else:
+                self.i += 1
+                return self.values[self.i]
```

Comment:
Because it is incrementing the index variable i after comparing it to the length, this will actually try to access an element that is after the last element. So instead you could do:

    if self.i > len(self.values) - 2: